### ISSUE TITLE
Slight discrepancy reproducing C2C (Enhanced + CSP) results on Sth-com

### ISSUE THREAD

**Issue by Shahzadnit** [RongchangLi/ZSCAR_C2C#13]
Hello,
First of all, thank you very much for releasing the code and dataset for the C2C paper. I have been working on reproducing the results on the Sth-com dataset using the provided config files.
Using the config c2c_enhance_vlm_vit_32_csp, I was able to successfully train both Vanilla and Enhanced versions. The results I obtained are close but slightly lower than those reported in the paper (Table 3).

📊 My Reproduced Results
C2C (Vanilla) + CLIP Text Encoder + CSP
 Best Unseen: 0.5157 | Best Seen: 0.4807 | HM: 0.4156 | AUC: 0.2254
C2C (Enhanced) + CLIP Text Encoder + CSP
 Best Unseen: 0.5327 | Best Seen: 0.4997 | HM: 0.4308 | AUC: 0.2422

📊 Paper Reported Results (Table 3)
C2C (Vanilla) + CLIP Text Encoder + CSP
 HM: 42.5 | AUC: 23.8
C2C (Enhanced) + CLIP Text Encoder + CSP
 HM: 44.3 | AUC: 25.5

⚙️ Config I Used
model:
  framework: vlm
  method: c2c_enhance
  text_encoding_manner: component
  backbone: "ViT-B/32"

text:
  text_input_type: template
  input_template_verb: "A verb of x"
  input_template_obj: "An object of x"
  learn_input: true
  learn_input_method: csp
  verb_length_type: natural
  ctx_init: true
  text_lr: 0.0001
  text_wd: 0.00001
  ctx_length: 10

data:
  dataset: sth-com
  dataset_path: "/Something_something_final/"
  num_frames: 8
  num_workers: 4

c2c:
  nlayers: 2
  fc_emb: 768,1024,1200
  feat_dim: 512
  emb_dim: 300
  relu: False

visual:
  pretrained: CLIP
  num_frames: 8
  adapt_star_layer: 6
  num_tadapter: 2
  num_workers: 4
  cosine_scale: 100
  visual_lr: 0.0005
  visual_wd: 0.0001

code_adapt:
  arch: vit
  aux_input: False
  ade_input: False

train:
  train_batch_size: 64
  gradient_accumulation_steps: 1
  seed: 0
  epochs: 50
  val_epochs_ts: 45
  warmup: 3
  epoch_start: 0
  save_path: '/ZSCAR_C2C/c2c_enhance_vlm_vit_32_csp/'
  load_model: False
  best_model_metric: AUC
  eval_every_n: 5
  save_every_n: 5
  aux_input: False
  ade_input: False
  cutmix_prob: 0.7
  beta: 1.0

test:
  eval_batch_size: 32
  open_world: False
  topk: 1
  text_encoder_batch_size: 36
  bias: 0.001
  pretrain: False
  load_model: YOUR_PATH

 Questions
1) Were the reported results in the paper averaged across multiple random seeds?
2) Are the coefficients α (for Lcomp), β (for Lind), and γ (for Lnew/Lcon) hard-coded anywhere? In the provided config I only see beta.
3) Is there any specific detail about hardware or other training settings (e.g., prompt initialization) that might explain the gap?
4)Should the number of training epochs be 30 (as mentioned in the supplementary) instead of 50 (as in the config)?

Thanks a lot for clarifying this! It will help ensure a faithful reproduction.

Best regards,
Shahzad Ahmad

**Comment by RongchangLi** [comment]
Hello—thanks so much for your interest. Here are the answers to your question: 
1. The results in the paper come from a single run. I performed ablations to determine the hyperparameters, as detailed in the supplementary material. 
2. The VLM training config files were reorganized for the GitHub release. In some experiments, I directly modified α, β, and γ in train_models.py—sorry I didn’t update this before uploading. The hyperparameters should be consistent with the descriptions in the supplementary material. 
3. VLM should be trained with fewer runs. CLIP-based paradigms converge faster than Visual model+text embedding models, so it should be 30 epochs.

**Comment by Shahzadnit** [comment]
Thank you very much for your reply and clarification regarding the training setup. I really appreciate the details you shared about the single-run results, the reorganization of configs, and the note that CLIP-based VLM settings should be trained for 30 epochs.

In our experiments, however, we consistently observed results that are about 3% lower in unseen classes compared to those reported in the paper, while seen accuracy remains close. For example, in the C2C (Enhanced) + CLIP + CSP setting:

Paper (Table 3): Unseen = 0.562, Seen = 0.499, HM = 0.443, AUC = 0.255

Our runs (multiple seeds): Unseen ≈ 0.533–0.545, Seen ≈ 0.495–0.500, HM ≈ 0.431–0.433, AUC ≈ 0.242–0.246

We carefully followed the supplementary hyperparameters (α=0.2, β=0.1, γ=0.1, CutMix p=0.7, 30 epochs, warmup=3, etc.), but the discrepancy in unseen accuracy remains.

Could you kindly confirm the exact values of α, β, and γ used in the released experiments?
And if possible, would you be able to share the pretrained checkpoints for the VLM setting (e.g., C2C Enhanced + CSP/CoOp/SPM)? That would help us verify whether the gap is due to subtle implementation details or environment differences.

Thanks again for your time and support.

**Comment by RongchangLi** [comment]
Hello! Sorry for the late reply. Since I’ve already graduated and didn’t back up the data on the previous server, I’m currently unable to provide the checkpoints. In my experiments at the time, γ and CutMix p had relatively large effects—I recommend trying smaller values, although the optimal values are γ = 0.1 and CutMix p = 0.7 in my experiments. In addition, the choice of optimizer, learning rate, and weight decay also has a significant impact. I suggest running an ablation study to pinpoint where the issue lies.
